ColorCoder.cpp

```diff
@@ -0,0 +1,39 @@
+#include "ColorCoder.h"
+#include <sstream>
+#include <iostream>
+
+namespace TeleCommColorCoder
+{
+    ColorPair ColorCoder::GetColorFromPairNumber(int pairNumber) 
+    {
+        int zeroBasedPairNumber = pairNumber - 1;
+        MajorColor majorColor = (MajorColor)(zeroBasedPairNumber / MinorColor::NUMBEROFMINORCOLORS);
+        MinorColor minorColor = (MinorColor)(zeroBasedPairNumber % MinorColor::NUMBEROFMINORCOLORS);
+        return ColorPair(majorColor, minorColor);
+    }
+
+    int ColorCoder::GetPairNumberFromColor(MajorColor major, MinorColor minor) 
+    {
+        return major * MinorColor::NUMBEROFMINORCOLORS + minor + 1;
+    }
+
+    void ColorCoder::PrintColorPairs()
+    {
+        std::ostringstream helpText;
+        helpText << "Number\t" << "Major Color\t" << "Minor Color" << std::endl;
+
+        for (int majorColor = WHITE; majorColor < NUMBEROFMAJORCOLORS; majorColor++)
+        {
+            for (int minorColor = BLUE; minorColor < NUMBEROFMINORCOLORS; minorColor++)
+            {
+                ColorPair pairToLog(static_cast<MajorColor>(majorColor), static_cast<MinorColor>(minorColor));
+
+                helpText << GetPairNumberFromColor(pairToLog.getMajor(),pairToLog.getMinor()) << "\t";
+                helpText << pairToLog.ToString() << std::endl;
+            }
+        }
+
+        std::cout << helpText.str();

```

Comment:
When we think of making this function testable, maybe we can return `helpText` from here and print it outside?
That will make it possible to assert on the return value of this function - removes some manual inspection effort to verify its working